### ISSUE TITLE
Using `clang` provided by Visual Studio causes issues

### ISSUE THREAD

**Issue by braydonk** [nvim-treesitter/nvim-treesitter#4861]
### Describe the bug

When using the version of `clang` provided by Visual Studio Build Tools, I got the `*.so is not a valid WIN32 application` errors that the [troubleshooting section](https://github.com/nvim-treesitter/nvim-treesitter/wiki/Windows-support#troubleshooting) of the Windows Support page mentions. I resolved this only by installing `llvm` through `scoop`. This should probably be called out in the Windows Support page, provided I did not do anything obviously wrong.

### To Reproduce

1. Install neovim with nvim-treesitter
2. Install Visual Studio (I was using 2019) and include "Clang build tools" in your installation
3. Ensure the `clang` binary at i.e. `C:\BuildTools\VC\Tools\Llvm\bin` (depends on your installation) is on your path such that nvim-treesitter can discover it as the compiler
4. Run `:TSInstall c`
5. Open a `.c` file, where you will run into the full `c.so is not a valid WIN32 application` error

### Expected behavior

It is possible that a `clang` version as old as what's in Visual Studio 2019 Build Tools simply will not work. It is v12. When I removed the Visual Studio `clang` from my path and instead installed `llvm` through `scoop` everything worked as expected.

I think the Windows Support page should call out that the version of `clang` with Visual Studio is known to cause problems. I don't know if this should be a callout in the `LLVM (Clang)` section of the page, or if it should go into Troubleshooting.

### Output of `:checkhealth nvim-treesitter`

```markdown
Everything was healthy except for the `*.so is not a valid WIN32 application` errors for every parser that was installed. After installing clang through scoop everything is healthy and working as expected once again.
```


### Output of `nvim --version`

```text
NVIM v0.10.0-dev-373+g07883940b                                                                                                                                                                                                        Build type: RelWithDebInfo                                                                                                                                                                                                             LuaJIT 2.1.0-beta3                                                                                                                                                                                                                     Compilation: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe /MD /Zi /O2 /Ob1  -W3 -wd4311 -wd4146 -DUNIT_TESTING -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_DEPRECATE -D_WIN32_WINNT=0x0602 -DMSWIN -DINCLUDE_GENERATED_DECLARATIONS -DNVIM_TS_HAS_SET_MAX_START_DEPTH -ID:/a/neovim/neovim/.deps/usr/include/luajit-2.1 -ID:/a/neovim/neovim/.deps/usr/include -ID:/a/neovim/neovim/build/src/nvim/auto -ID:/a/neovim/neovim/build/include -ID:/a/neovim/neovim/build/cmake.config -ID:/a/neovim/neovim/src                                                                                                                                                                                                                                                                                                                                                                                 system vimrc file: "$VIM\sysinit.vim"                                                                                                                                                                                                 fall-back for $VIM: "C:/Program Files (x86)/nvim/share/nvim"                                                                                                                                                                                                                                                                                                                                                                                                                Run :checkhealth for more info
```


### Additional context

_No response_

**Comment by braydonk** [comment]
Oops, I didn't realize quite how GitHub Wikis work, and submitted the edit with the callout :sweat_smile: I thought it opened a PR or something. Would someone be able to check that the edit makes sense? If it's sufficient then the issue can be closed.

**Comment by clason** [comment]
Looks good to me. Maybe add a "known good" version as well?

**Comment by braydonk** [comment]
I know that when I installed `llvm` through scoop, it installed `clang` 16.0.4. I can put that specifically, but not sure if that's too anecdotal?

**Comment by theHamsta** [comment]
Can you rename `c.so` to `c.dll` and right click to get the file properties? Would it be recognized as a DLL at all? You would get the same error when you picked the wrong architecture (64bit vs 32bit). Can a third party tool like https://github.com/lucasg/Dependencies give more info?

**Comment by braydonk** [comment]
Thank you for the suggestion @theHamsta! Your suggestion about the architecture turned out to be correct; turns out installing `clang` through Visual Studio provides `x86` and `x64` versions of `clang` and they are in different paths. I had configured the wrong one.

I removed the edit to the page that I mentioned earlier and instead added a section about how to properly use the `clang` install provided by Visual Studio. I think I will close this issue, but let me know if the Wiki edit needs some more detail.

Appreciate the help!
